src/hammer-tech/stackup.py

```diff
@@ -121,48 +126,33 @@ class Metal(NamedTuple('Metal', [
     """
     __slots__ = ()
 
-    @property
-    def grid_unit(self) -> float:
-        """
-        Return the manufacturing grid unit.
-
-        TODO: this assumes a manufacturing grid of 0.001
-        """
-        return 0.001
-
-    def snap(self, num: float) -> float:
+    @staticmethod
+    def from_setting(grid_unit: Decimal, d: dict) -> "Metal":
         """
-        Snap a number to the grid unit.
-
-        TODO: internally represent numbers as integers or fixed-point to
-        obviate the need for this (see #319).
+        Return a Metal object from a dict with keys "name", "index", "direction", "min_width", "pitch", "offset", and "power_strap_widths_and_spacings"
 
-        :param num: Number to snap
-        :return: Number snapped to grid_unit
+        :param grid_unit: The manufacturing grid unit in nm
```

Comment:
Did we decide on `nm` or `um`? IIRC many of my CAD scripts units are in `um`. Maybe we should make use of `hammer_units`

Reply:
um, this is just a typo.

Reply:
Hmm `hammer_units` is an interesting idea.

Reply:
Sorry for the blast of comments on this one. Maybe we can convert to `hammer_units` in a future PR? I have filed #376 to convert all our internal units to decimals- maybe it makes sense to combine these two changes?

Reply:
Sounds good; let's file an issue?

Reply:
Issue is #378